actionpack/lib/action_dispatch/http/request.rb

```diff
@@ -180,11 +180,28 @@ def http_auth_salt
       get_header "action_dispatch.http_auth_salt"
     end
 
-    def show_exceptions? # :nodoc:
+    def show_exceptions?(exception) # :nodoc:
       # We're treating `nil` as "unset", and we want the default setting to be
-      # `true`. This logic should be extracted to `env_config` and calculated
+      # `:all`. This logic should be extracted to `env_config` and calculated
       # once.
-      !(get_header("action_dispatch.show_exceptions") == false)
+      config = get_header("action_dispatch.show_exceptions")
+
+      # For backwards compatibility.
+      if config == true
+        ActiveSupport::Deprecation.warn("Setting action_dispatch.show_exceptions to true is deprecated. Set to :all instead.")
+        config = :all
+      elsif config == false
+        ActiveSupport::Deprecation.warn("Setting action_dispatch.show_exceptions to false is deprecated. Set to :none instead.")
+        config = :none
+      end
+
+      if config == :none
+        false
+      elsif config == :rescuable
+        ActionDispatch::ExceptionWrapper.rescue_responses.include?(exception.class.name)
```

Comment:
Request should not know about the existence of the `ExceptionWrapper`. Isn't the exception already wrapped when we call this method? I think it can be since we wrap it in `invoke_interceptors` and also later in `render_exception`. We should just ask to the exception if it is a `rescue_response?`. I'd even move this entire implementation out of `Request` since it belongs more to the `ExceptionWrapper` than to the request. We should just read the `get_header("action_dispatch.show_exceptions")` from the request.

Reply:
@rafaelfranca Thanks for the review and sorry for the late reply. I have applied this suggestion in the latest revision. Please let me know if it is what you had in mind. All additional feedback is welcome.